src/Form/Extension/Product/ProductTranslationTypeExtension.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Setono\SyliusRedirectPlugin\Form\Extension\Product;
+
+use Sylius\Bundle\ProductBundle\Form\Type\ProductTranslationType;
+use Sylius\Component\Core\Model\ProductTranslationInterface;
+use Symfony\Component\Form\AbstractTypeExtension;
+use Symfony\Component\Form\Extension\Core\Type\CheckboxType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+
+final class ProductTranslationTypeExtension extends AbstractTypeExtension
```

Comment:
You should make an abstract class that does the heavy lifting in this class and then just extend that class to provide the method `getExtendedTypes`. This way it's much easier for others to reuse this abstract class to do this on i.e. taxons or other resources